src/js/view/view.js

```diff
@@ -875,15 +875,29 @@ function View(_api, _model) {
 
             // Resize within MAX_FLOATING_WIDTH×MAX_FLOATING_HEIGHT bounds, never enlarge.
             const { width, height } = _this.getSafeRegion(false);
-            const ratio = Math.min(1, MAX_FLOATING_WIDTH / width, MAX_FLOATING_HEIGHT / height);
-            _this.resize(width * ratio, height * ratio, true);
+            const floatingWidth = Math.min(width, MAX_FLOATING_WIDTH);
+            const floatingHeight = Math.min(height * floatingWidth / width, MAX_FLOATING_HEIGHT);
+            _this.resize(floatingWidth, floatingHeight);
 
             _resizeOnFloat = false;
         } else if (isVisible) {
             _this.stopFloating();
         }
     }
 
+    this.resizeFloatingPlayer = function() {
+        if (floatingPlayer && !_resizeOnFloat) {
+            const rect = bounds(_playerElement);
```

Comment:
we're already calling `bounds` in `updateBounds`, which calls `element.getBoundingClientRect();`. I believe there are performance issues with calling this since in some browsers it forces a new layout calculation. We should look for a way to minimize the calls to `bounds`